### PR TITLE
Fix #3997 - GbXMLReverseTranslator applies incorrect scaling to windows when unit isn't meter

### LINKED ISSUE #3997: 3.0.1-beta1: GbXMLReverseTranslator applies incorrect scaling to windows
Issue overview
--------------

With `lengthUnit="Feet"`, the windows are incorrectly placed.

gbxml scaling issue

Reported by @antonszilasi

## Current Behavior

Wrong scaling for windows

## Expected Behavior

Windows should be placed correctly, like they are in 2.9.1.

## Steps to Reproduce

I will make a unit test for this.

## Possible Solution

Move from QXml to pugixml missed a conversion factor `m_lengthMutiplier` as it previously existed here: https://github.com/NREL/OpenStudio/blame/e65085e23f2792b8726bdd6f3f143c31651d136a/openstudiocore/src/gbxml/ReverseTranslator.cpp#L559

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of OpenStudio (if using an intermediate build, include SHA): 3.0.1-beta1, c710464d0534eba5f2b52b3ac9b01a17c4a61a6e

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

- Fix #3997
- GbXMLReverseTranslator applies incorrect scaling to windows when unit isn't meter

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [x] gbXML API Changes / Additions
 - [x] gbXML API methods are tested (in `src/gbxml/test`)
 - [x] All new and existing tests passes

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/gbxml/Test/ReverseTranslator_GTest.cpp [hunk 1]
```diff
@@ -525,3 +526,20 @@ TEST_F(gbXMLFixture, ReverseTranslator_3951_Surface)
...
+  ASSERT_TRUE(_model);
+
+  // Check the SubSurface is contained on the same plane as its Surface
+  auto _subSurfaces = _model->getConcreteModelObjects<SubSurface>();
+  ASSERT_EQ(1u, _subSurfaces.size());
+  SubSurface ss = _subSurfaces[0];
+  ASSERT_TRUE(ss.surface());
+  Surface s = ss.surface().get();
+  EXPECT_TRUE(ss.plane().equal(s.plane()));
+}
```
Comment (jmarrec) [inline src/gbxml/Test/ReverseTranslator_GTest.cpp:545]: New Unit test, loads a MCVE gbxml

File: resources/gbxml/3997_WindowScaling_bug.xml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+            </CartesianPoint>
+            <CartesianPoint>
+              <Coordinate>139.2895</Coordinate>
+              <Coordinate>-316.4557</Coordinate>
+              <Coordinate>1.8</Coordinate>
+            </CartesianPoint>
+          </PolyLoop>
+        </PlanarGeometry>
+      </Opening>
+    </Surface>
```
Comment (jmarrec) [inline resources/gbxml/3997_WindowScaling_bug.xml:80]: The surface and subsurface we care about. Rest of the gbxml file is just the floor/ceilings and other walls, so the space is bounded.

File: resources/CMakeLists.txt [hunk 3]
```diff
@@ -285,6 +285,7 @@ SET(gbxml_resources_src
   gbxml/TwoStoryOffice_Trane.xml
   gbxml/ZNETH.xml
   gbxml/3951_Geometry_bug.xml
+  gbxml/3997_WindowScaling_bug.xml
```
Comment (jmarrec) [inline resources/CMakeLists.txt:288]: Add new xml test file to the resources CMakeLists.txt

**Comment by jmarrec** [comment]
Don't mind CI, Windows machine is doing something weird again. Initial commit 2f1eda2 shows the new unit test `ReverseTranslator_3997_WindowScaling` failing, which then passes with the fix in the second commit a5804b8. cf https://ci.commercialbuildings.dev/blue/organizations/jenkins/openstudio-incremental-develop/detail/PR-3998/2/tests

image

**Comment by jmarrec** [comment]
I rebuilt OSApp with the SDK from this PR. The before/after is below, and shows that it fixes both #3951 (surface types) and #3997

### With openstudioapplication-1.0.0 (OS 3.0.0):

image


###  With this PR

image


@antonszilasi FYI.

**Comment by antonszilasi** [comment]
@tijcolem @jmarrec  thanks I confirm this is fixed with the nrel/openstudio:3.0.1-rc2 image

**Pull request merged**